src/ray/raylet/task_dependency_manager.cc

```diff
@@ -297,6 +298,26 @@ void TaskDependencyManager::TaskCanceled(const TaskID &task_id) {
   }
 }
 
+void TaskDependencyManager::CleanupForDriver(const std::unordered_set<TaskID> &task_ids) {
+  for (auto it = task_ids.begin(); it != task_ids.end(); it++) {
+    task_dependencies_.erase(*it);
+    required_tasks_.erase(*it);
+    pending_tasks_.erase(*it);
+  }
+
+  for (auto it = required_objects_.begin(); it != required_objects_.end();) {
```

Comment:
It seems to me that this could potentially be a perf problem. Because the length of `required_objects_` won't be short, I guess. 
I'm not sure if it's worth it to add an index here, but at least we can add a TODO here to remind us it may need to be optimized. 
Also we can return earlier when `task_ids` is empty, which is likely to be true in many cases.

Reply:
This is a good point. Thanks!